app/src/main/java/tech/pauly/findapet/shared/WrapViewPager.java

```diff
@@ -0,0 +1,35 @@
+package tech.pauly.findapet.shared;
+
+import android.content.Context;
+import android.support.v4.view.ViewPager;
+import android.util.AttributeSet;
+import android.view.View;
+
+public class WrapViewPager extends ViewPager {
+
+    public WrapViewPager(Context context) {
+        super(context);
+    }
+
+    public WrapViewPager(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    @Override
+    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+        int height = 0;
+        int currItem = getCurrentItem();
+        if (currItem < 0) { currItem = 0; }
```

Comment:
Don't like this solution, it still doesn't work like I want. I need the height to change on page change events, and I'm not getting onMeasure called again. Not too familiar with ViewPager details, so investigating this later.